configurable-prompter.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const inquirer = require("inquirer");
+const _ = require("lodash");
+const findConfig = require("find-config");
+
+function getConfig (filename) {
+    const pkg = findConfig.require("package.json");
+    const configFile = _.get(pkg, "config.cz-ovh-commit.config", filename);
+    console.log("Using config file: ", configFile);
+    return Object.assign({}, require(filename), require(configFile));
+}
+
+module.exports = function (configFilename) {
+    return {
+        prompter (cz, commit) {
+            const config = getConfig(configFilename);
+
+            inquirer
+                .prompt([
+                    {
+                        type: "list",
+                        name: "type",
+                        message: "Please select the TYPE of your commit: ",
+                        choices: config.TYPES,
+                        when: true
+                    },
+                    {
+                        type: "list",
+                        name: "type",
+                        message: (answers) => `Please select the kind of ${answers.type}: `,
+                        choices: (answers) => config.SUB_TYPES[answers.type],
+                        when: (answers) => config.SUB_TYPES[answers.type] != null && config.SUB_TYPES[answers.type].length > 0,
+                        filter: (input) => _.trim(input)
+                    },
+                    {
+                        type: "input",
+                        name: "scope",
+                        message: "Please denote the SCOPE of the commit: (optional)",
+                        when: (answers) => _.includes(config.useScope, answers.type),
+                        filter: (input) => _.trim(input)
+                    },
+                    {
+                        type: "input",
+                        name: "header",
+                        message: "Please write a HEADER: (mandatory)",
+                        when: (answers) => _.includes(config.useHeader, answers.type),
+                        validate: (input) => input && _.trim(input).length < config.headerLength && _.trim(input).length !== 0 ? true : `Message should not be empty and should be shorter than ${config.headerLength} characters`,
```

Comment:
could be simplified (an empty string evaluates to false) and lisibility could be improved. this line is too long